### ISSUE TITLE
Allow customizing grpc service code generators in Gradle

### ISSUE THREAD

**Issue by jvican** [scalapb/ScalaPB#891]
I'm writing a ScalaPB service code generator for Monix. I have the code generator working, so now I want to add it to my gradle build.
  
However, after having a look at https://github.com/scalapb/gradle-demo and https://github.com/google/protobuf-gradle-plugin, I've come to the conclusion that there's no way to customize the service generators that `protoc-gen-scala` uses. Or, at least, I don't know how to do that.
  
Could we add support for this feature in scalapb? Does this mean that we would need to generate a `protoc-gen-scala` that is a binary that has my custom service generator as a dependency? Ideally, `protoc-gen-scala` would have an option to pass the plugin artifact as scalapbc has and run it through system process (given that it's not built inside the binary).

**Comment by thesamet** [comment]
Can you share more about how you set up your code generator? Normally, your code generator would run next to protoc-gen-scala, generating additional files, rather than customizing the output of protoc-gen-scala.

Were you able to assemble your plugin as an executable jar like protoc-gen-scala? This way you'll be able to add it like this:

```
  plugins {
    scalapb {
        artifact = (org.gradle.nativeplatform.platform.internal.DefaultNativePlatform.getCurrentOperatingSystem().isWindows()) ?
            "com.thesamet.scalapb:protoc-gen-scalapb:${scalapbVersion}:windows@bat" :
            "com.thesamet.scalapb:protoc-gen-scalapb:${scalapbVersion}:unix@sh"
    }
    monix {
       artifact = ...
    }
  }

If you are building your plugin in sbt, I wrote an [sbt plugin](https://github.com/thesamet/sbt-protoc-gen-project) that makes it easy to develop, build and test code generators. You can see how it's used in zio-grpc or in [scalapb-validate](https://github.com/scalapb/scalapb-validate/)

I didn't completely follow about the ScalaPB feature proposal, if the solution above doesn't cover it - can you describe more what that feature would do?

**Comment by thesamet** [comment]
Hi @jvican,  is there anything still needed for this issue?

**Comment by jvican** [comment]
Thanks @thesamet, that information was useful and pointed me in the right direction. What I didn't seem to understand is that Monix would become an independent protobuf generator. I did some experimentation on my own and figured out a way to publish the protobuf binaries with the expected classifiers in Gradle, the build tool I'm using to define the codegen project. After that, I was able to get it working.
